src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/Models/ReindexJobQueryStatusConverter.cs

```diff
@@ -0,0 +1,44 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics.CodeAnalysis;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Reindex.Models
+{
+    /// <summary>
+    /// Class to hold metadata for one query of a reindex job
```

Comment:
>Class to hold metadata for one query of a reindex job [](start = 8, length = 53)

nit: Can we change it to:
JsonConverter to handle change <see cref="ReindexJobRecord.QueryList"> from ConcurrentBag to ConcurrentDictionary. 
<remarks>
For backcompat and fact what we don't need values in dictionary and only key uniqueness, we present it as array and not dictionary.
</remarks>

Reply:
Agreed, some comments would be helpful to explain this class.